src/components/results/loadflow/load-flow-result.tsx

```diff
@@ -217,45 +271,64 @@ export const LoadFlowResult: FunctionComponent<LoadflowResultProps> = ({
             overloadedEquipment.limitType === LimitTypes.HIGH_VOLTAGE ||
             overloadedEquipment.limitType === LimitTypes.LOW_VOLTAGE
     );
+
+    useEffect(() => {
+        if (voltageViolations && isFetchComplete) {
+            setIsVoltageViolationReady(true);
+        }
+    }, [voltageViolations, isFetchComplete]);
+
     const renderLoadFlowVoltageViolations = () => {
         const message = getNoRowsMessage(
             messages,
             voltageViolations,
-            loadFlowStatus
+            loadFlowStatus,
+            isVoltageViolationReady
         );
 
         const rowsToShow = getRows(voltageViolations, loadFlowStatus);
         return (
-            <CustomAGGrid
-                rowData={rowsToShow}
-                defaultColDef={defaultColDef}
-                enableCellTextSelection={true}
-                columnDefs={loadFlowVoltageViolationsColumns}
-                onGridReady={onGridReady}
-                getRowStyle={getRowStyle}
-                overlayNoRowsTemplate={message}
-            />
+            <>
+                <Box sx={{ height: '4px' }}>
+                    {openLoaderVoltageTab && <LinearProgress />}
+                </Box>
+                <CustomAGGrid
+                    rowData={rowsToShow}
+                    defaultColDef={defaultColDef}
+                    enableCellTextSelection={true}
+                    columnDefs={loadFlowVoltageViolationsColumns}
+                    onGridReady={onGridReady}
+                    getRowStyle={getRowStyle}
+                    overlayNoRowsTemplate={message}
+                />
+            </>
         );
     };
 
     const renderLoadFlowResult = () => {
         const message = getNoRowsMessage(
             messages,
             result?.componentResults,
-            loadFlowStatus
+            loadFlowStatus,
+            result?.componentResults ? true : false
```

Comment:
```suggestion
            !!result?.componentResults
```

Reply:
yeah, I've seen this suggestion by the IDE but for me it's not so clear at first glance, I'd rather make it a few characters longer but human-readable